cmd/aro/monitor.go

```diff
@@ -59,7 +61,20 @@ func monitor(ctx context.Context, log *logrus.Entry) error {
 		return err
 	}
 
-	key, err := _env.GetBase64Secret(ctx, env.EncryptionSecretName)
+	rpKVAuthorizer, err := _env.NewRPAuthorizer(_env.Environment().ResourceIdentifiers.KeyVault)
+	if err != nil {
+		return err
+	}
+
+	// TODO: should not be using the service keyvault here
```

Comment:
Why? How monitor would get kubeconfig to access cluster? Or you expect us to maintain separate table with different kubeconfig for monitor?

Reply:
There's some initial detail in the permission separation architecture doc.  The monitor should run as a separate identity and there will be a database token service in yet another identity.  Either everyone who needs access to the encryption secret will access it using their identity from a new shared key vault, or they will get it via REST API from the database token service.  The details depend on how we want to build towards encryption secret rotation.